src/nbl/video/CVulkanPhysicalDevice.h

```diff
@@ -603,6 +664,18 @@ class CVulkanPhysicalDevice final : public IPhysicalDevice
                 m_features.shaderSubgroupExtendedTypes = shaderSubgroupExtendedTypesFeaturesKHR.shaderSubgroupExtendedTypes;
             }
 
+            if (isExtensionSupported(VK_EXT_SHADER_VIEWPORT_INDEX_LAYER_EXTENSION_NAME))
+            {
+                m_properties.limits.shaderOutputViewportIndex = true;
+                m_properties.limits.shaderOutputLayer = true;
+            }
+            else if (instanceApiVersion >= VK_MAKE_API_VERSION(0, 1, 2, 0))
+            {
+                m_properties.limits.shaderOutputViewportIndex = vulkan12Features.shaderOutputViewportIndex;
+                m_properties.limits.shaderOutputLayer = vulkan12Features.shaderOutputLayer;
+            }
```

Comment:
I think this is the most complex logic we've had in the vulkan backend :D, I just read the spec and It seems that you're doing the right thing, In GL this kinda logic is everywheere :D